fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1003,8 +1003,18 @@ public Void visitFunctionCall(FunctionCallExpr node, Scope scope) {
 
             // throw exception direct
             checkFunction(fnName, node, argumentTypes);
-
-            if (fnName.equals(FunctionSet.COUNT) && node.getParams().isDistinct()) {
+            if (fnName.equalsIgnoreCase("typeof") && argumentTypes.length == 1) {
+                // For the typeof function, the parameter type of the function is the result of this function.
+                // At this time, the parameter type has been obtained. You can directly replace the current 
+                // function with StringLiteral. However, since the parent node of the current node in ast 
+                // cannot be obtained, this cannot be done directly. Replacement, here the StringLiteral is 
+                // stored in the parameter of the function, so that the StringLiteral can be obtained in the 
+                // subsequent rule rewriting, and then the typeof can be replaced.
+                Type originType = argumentTypes[0];
+                argumentTypes[0] = Type.STRING;
+                fn = new Function(new FunctionName("typeof_internal"), argumentTypes, Type.STRING, false);
+                node.setParams(Arrays.asList(new StringLiteral(originType.toTypeString())));
```

Comment:
could we move the originType.toTypeString() logic to ScalarOperatorFunctions? Then we don't need to add the setParams method to FunctionCallExpr class

Reply:
That's a good idea. I'll fix it.

Reply:
But ScalarOperatorFunctions does not seem to be able to handle complex types such as array, struct, map, etc.

Reply:
> could we move the originType.toTypeString() logic to ScalarOperatorFunctions? Then we don't need to add the 
Moving it is difficult. We need to modify the logic in FoldConstantsRule、ScalarOperatorEvaluator and ScalarOperatorFunctions. Maybe I can move the relevant logic to the current location to avoid modifying the code of the FunctionCallExpr class. What do you think it? @kangkaisen

Reply:
how about add a speical process in `SqlToScalarOperatorTranslator.visitFunctionCall()`. We transform it to `typeof_internal` and set the new argument here. @Seaven  @kangkaisen

Reply:
"Maybe I can move the relevant logic to the current location to avoid modifying the code of the FunctionCallExpr class.  " I think this way is OK.